peartree/parallel.py

```diff
@@ -165,7 +175,7 @@ def generate_all_observed_edge_costs(trips_and_stop_times: pd.DataFrame
                 dir_mask = (tst_sub.direction_id == direction)
                 tst_sub_dir = tst_sub[dir_mask]
             else:
-                tst_sub_dir = tst_sub
+                tst_sub_dir = tst_sub.copy()
```

Comment:
Why the removal of the `.copy()`? Without examining it deeply, shouldn't we defensively employ `.copy()` to avoid upstream impacts? It's removal does not seem to be relevant to handling the `direction_id` column being dropped.

Reply:
Ignore this. I am so sorry.